code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/resource/ServiceDiscoveryConfigItem.java

```diff
@@ -0,0 +1,131 @@
+    public static final ServiceDiscoveryConfigItem EXTERNALLINKS = new ServiceDiscoveryConfigItem(
+            "instanceLinks",
+            "external_links", true, true);
+    public static final ServiceDiscoveryConfigItem LINKS = new ServiceDiscoveryConfigItem(null,
+            "links", false, true);
+
+    // RANCHER PARAMETERS
+    public static final ServiceDiscoveryConfigItem REGISTRYCREDENTIALID = new ServiceDiscoveryConfigItem("registryCredentialId",
+            "registryCredentialId", true, false);
+    public static final ServiceDiscoveryConfigItem SCALE = new ServiceDiscoveryConfigItem("scale", "scale",
+            false, false);
+
+    /**
+     * Name as it appears in docker-compose file
+     */
+    private String composeName;
+
+    /**
+     * Name as it appears in rancher (can be diff from what defined in a config)
+     */
+    private String rancherName;
+
+    /**
+     * Defines whether the property is the launch config property
+     */
+    private boolean isLaunchConfigItem;
+
+    private boolean isDockerComposeProperty;
+
+    public ServiceDiscoveryConfigItem(String rancherName, String composeName, boolean isLaunchConfigItem,
+            boolean isDockerComposeProperty) {
+        super();
+        this.rancherName = rancherName;
+        this.composeName = composeName;
+        this.isLaunchConfigItem = isLaunchConfigItem;
+        this.isDockerComposeProperty = isDockerComposeProperty;
+        supportedServiceConfigItems.add(this);
+    }
+
+    public ServiceDiscoveryConfigItem() {
+    }
+
+    public ServiceDiscoveryConfigItem(String rancherName, String composeName) {
+        this(rancherName, composeName, true, true);
+    }
+
+    public static List<ServiceDiscoveryConfigItem> getSupportedLaunchConfigItems() {
+        return supportedServiceConfigItems;
+    }
+
+    public String getComposeName() {
+        return composeName;
+    }
+
+    public boolean isLaunchConfigItem() {
+        return isLaunchConfigItem;
+    }
+
+    public String getRancherName() {
+        return rancherName;
+    }
+
+    public static ServiceDiscoveryConfigItem getServiceConfigItemByComoposeName(String externalName) {
+        for (ServiceDiscoveryConfigItem serviceItem : supportedServiceConfigItems) {
+            if (serviceItem.getComposeName().equalsIgnoreCase(externalName)) {
+                return serviceItem;
+            }
+        }
+        return null;
+    }
+
+    public static ServiceDiscoveryConfigItem getServiceConfigItemByRancherName(String internalName) {
+        for (ServiceDiscoveryConfigItem serviceItem : supportedServiceConfigItems) {
+            if (serviceItem.getRancherName() != null && serviceItem.getRancherName().equalsIgnoreCase(internalName)) {
+                return serviceItem;
+            }
+        }
+        return null;
+    }
+
+    public boolean isDockerComposeProperty() {
```

Comment:
Super minor.  Can we move this above the static functions so it's grouped with the rest of the properties?